src/MSMToolsCluster.jl

```diff
@@ -0,0 +1,259 @@
+using LinearAlgebra
+using Arpack
+using Statistics
+using SparseArrays
+using LightGraphs
+using Optim
+
+# NOTE: Reimplementation in Julia for select functions from MSM Tools (referenced in DeepTime)
+# https://github.com/markovmodel/msmtools
+# https://github.com/deeptime-ml/deeptime/tree/a6ac0b93a55d688fe8f3af119680105763366220/deeptime/markov/tools
+
+# Reimplementation of PCCA+ spectral clustering method with optimized memberships
+# https://github.com/deeptime-ml/deeptime/blob/a6ac0b93a55d688fe8f3af119680105763366220/deeptime/markov/tools/analysis/dense/_pcca.py#L288
+function pcca_memberships(P::Matrix{Float64}, m::Int, π::Vector{Float64}=nothing)
+    n = size(P, 1)
+    if m > n
+        throw(ArgumentError("Number of metastable states m=$m exceeds number of states of transition matrix n=$n"))
+    end
+    println("P: $P")
+    if !is_transition_matrix(P)
+        throw(ArgumentError("Input matrix is not a transition matrix."))
+    end
+    if isnothing(π)
+        if ndims(π) > 1
+            throw(ArgumentError("Expected 1D stationary distribution array"))
+        elseif length(π) != n
+            throw(ArgumentError("Stationary distribution must be defined on entire space, piecewise if the transition matrix " *
+            "has multiple connected components. It covered $(length(π)) != $n states."))
+        end
+    end
+
+    chi = zeros(n, m)
+    components = connected_sets(P)
+    closed_components = []
+    transition_states = []
+    for component in components
+        rest = setdiff(1:n, component)
+        # if component is closed, store with positive equilibrium distribution
+        if sum(P[component, rest]) == 0
```

Comment:
I would use the function `isapprox` to test if the sum is zero since,
1. The actual sum could be very close to zero but not exactly due to floating point error in which case the comparison fails
2. Comparing a Float (the result of the sum) with an Int would result in an implicit conversion

Point 2. is less important, and could be made even more "clever" with the use of types. The main point is 1.

Reply:
fixed in 70820c33962fe449bfbb4c417bda2628989df0b9